protoarray/src/main/java/tech/pegasys/teku/protoarray/ProtoArrayBuilder.java

```diff
@@ -16,74 +16,69 @@
-  private Checkpoint finalizedCheckpoint;
-  private Optional<ProtoArraySnapshot> protoArraySnapshot = Optional.empty();
+  public ProtoArray build() {
+    checkNotNull(justifiedEpoch, "Justified epoch must be supplied");
+    checkNotNull(finalizedEpoch, "finalized epoch must be supplied");
+    return new ProtoArray(pruneThreshold, justifiedEpoch, finalizedEpoch, initialEpoch, nodes);
+  }
 
-  public static ProtoArray fromAnchorPoint(final AnchorPoint anchor) {
-    final ProtoArray protoArray =
-        new ProtoArrayBuilder()
-            .justifiedCheckpoint(anchor.getCheckpoint())
-            .finalizedCheckpoint(anchor.getCheckpoint())
-            .build();
-    protoArray.onBlock(
-        anchor.getBlockSlot(),
-        anchor.getRoot(),
-        anchor.getParentRoot(),
-        anchor.getStateRoot(),
-        anchor.getEpoch(),
-        anchor.getEpoch());
-    return protoArray;
+  public ProtoArrayBuilder pruneThreshold(final int pruneThreshold) {
+    this.pruneThreshold = pruneThreshold;
+    return this;
   }
 
-  public ProtoArrayBuilder anchor(final Optional<Checkpoint> anchor) {
-    this.anchor = anchor;
+  public ProtoArrayBuilder initialEpoch(final UInt64 initialEpoch) {
+    this.initialEpoch = initialEpoch;
     return this;
   }
 
   public ProtoArrayBuilder justifiedCheckpoint(final Checkpoint justifiedCheckpoint) {
-    this.justifiedCheckpoint = justifiedCheckpoint;
+    checkNotNull(justifiedCheckpoint, "Justified checkpoint must be supplied");
+    this.justifiedEpoch(justifiedCheckpoint.getEpoch());
     return this;
   }
 
   public ProtoArrayBuilder finalizedCheckpoint(final Checkpoint finalizedCheckpoint) {
-    this.finalizedCheckpoint = finalizedCheckpoint;
+    checkNotNull(finalizedCheckpoint, "finalized checkpoint must be supplied");
+    this.finalizedEpoch(finalizedCheckpoint.getEpoch());
     return this;
   }
 
-  public ProtoArrayBuilder protoArraySnapshot(
-      final Optional<ProtoArraySnapshot> protoArraySnapshot) {
-    this.protoArraySnapshot = protoArraySnapshot;
+  public ProtoArrayBuilder justifiedEpoch(final UInt64 justifiedEpoch) {
+    this.justifiedEpoch = justifiedEpoch;
     return this;
   }
 
-  public ProtoArray build() {
-    checkNotNull(justifiedCheckpoint, "Justified checkpoint must be supplied");
-    checkNotNull(finalizedCheckpoint, "Finalized checkpoint must be supplied");
+  public ProtoArrayBuilder finalizedEpoch(final UInt64 finalizedEpoch) {
+    this.finalizedEpoch = finalizedEpoch;
+    return this;
+  }
 
-    // If no anchor is explicitly set, default to zero (genesis epoch)
-    final UInt64 anchorEpoch =
-        anchor.map(Checkpoint::getEpoch).orElse(UInt64.valueOf(Constants.GENESIS_EPOCH));
+  public ProtoArrayBuilder initialCheckpoint(final Optional<Checkpoint> initialCheckpoint) {
+    if (initialCheckpoint.isPresent()) {
+      initialEpoch(initialCheckpoint.get().getEpoch());
+    } else {
+      initialEpoch(SpecConstants.GENESIS_EPOCH);
+    }
```

Comment:
If you want to be fancy you could do:
```suggestion
    initialCheckpoint.ifPresentOrElse(
       checkpoint -> initialEpoch(checkpoint.getEpoch()),
       () -> initialEpoch(SpecConstants.GENESIS_EPOCH));
```

I'd probably have to see what spotless does to that to decide if it winds up being better or not 🤣